module-sys/SystemManager/GovernorSentinelOperations.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "GovernorSentinelOperations.hpp"
+
+namespace sys
+{
+
+    void for_each_governed_sentinel(GovernorSentinelsVector &sentinels, governed_callback &foo)
```

Comment:
Why underscore style? 
The naming as well doesn't imply what it really does. Maybe cleanInvalidSentinels/checkSentinels or smth like this.

Reply:
1. style: I followed the stl style here because it made sense to me that if normal for each is written `for_each` and if I would extend it to for each with if it would be `for_each_if` then here it would be `for_each_governed...` 
2. functionality: `cleanInvalidSentinels` would definitely be wrong. Please see that it's first and foremost for_each functionality with additional overzealous sanity check and invalid data removal. It's well documented in header and you can check how it works in tests :) 

I can change it to camel case and name it whatever you want, just tell me what would be your exact wishes :)

Reply:
@SP2FET what part doesn't comply with our guideline? I didn't find a word about camel case in it

Reply:
I get the stl style motivation and I accept it. As for guidelines, we prefer camelCase in most cases, however we don't force using it everywhere (if it is justified). Please consider moving cleaning of dangling and expired sentinels to another method and run it separately with clear intent as those checks are needed, however it is not obvious that those are done in a for_each type of method.

Reply:
I know, I was there enforcing it in the first place and i.e. promoting automatic hooks with @rwicik :) to be completely honest we might revisit automatic style checks for functions and enums.
Saying that - these have documentation in header and I thought are trivial enough but sure, I'll move the `erase` line to another function